src/components/thumbnail/thumbnail-document-item.tsx

```diff
@@ -2,13 +2,15 @@ import * as React from "react";
 import { CanvasComponent } from "../document/canvas";
 import { DocumentModelType } from "../../models/document/document";
 import { observer } from "mobx-react";
+import { IStores } from "../../models/stores/stores";
 
 interface IProps {
   dataTestName: string;
   canvasContext: string;
   document: DocumentModelType;
   scale: number;
   captionText: string;
+  stores: IStores;
```

Comment:
I'd rather keep this component as "dumb" as possible. The need for `stores` could be eliminated by passing in a `role` parameter instead which could be `primary-doc` or `secondary-doc` and then the caller would be responsible for accessing the keys in the workspace/stores.